### PR TITLE
Add data explorer

### PR DESCRIPTION
**Pull request by ChristiaensBert**
Add data_explorer folder with implementation of the Subset explorer UI.

The docker image with this logic is built with `scripts/build_explorer.sh` and pushed to the `ghcr.io/ml6team/data_explorer`.

In the `.toml` file, a script is added to run this docker image. When installing `Fondant`, which calls the cli.py file.

This script can be ran locally with:

```
fondant-explorer --source LOCAL_FOLDER
```

First merge Philippe and Georges their PRs

### CODE REVIEW

File: components/pii_redaction/src/gibberish_data/big.txt
Comment (ChristiaensBert) [inline components/pii_redaction/src/gibberish_data/big.txt:1]: @NielsRogge This came when merging your branch into mine. Is this needed or should I delete it?

Comment (RobbeSneyders) [reply components/pii_redaction/src/gibberish_data/big.txt:1]: Can be deleted

File: fondant/cli.py
Comment (RobbeSneyders) [inline fondant/cli.py:71]: This swallows all errors, which makes it hard to debug why something is not working.

File: fondant/cli.py
Comment (RobbeSneyders) [inline fondant/cli.py:34]: It's not clear to me what is expected as the source directory here. Can we add a better description (and maybe better name?)

And can we make this work with relative paths as well?

File: examples/pipelines/starcoder/build_images.sh
Comment (RobbeSneyders) [inline examples/pipelines/starcoder/build_images.sh:1]: I don't think these changes should be in here.

File: examples/pipelines/starcoder/pipeline.py
Comment (RobbeSneyders) [inline examples/pipelines/starcoder/pipeline.py:1]: I don't think these changes should be in here.

File: scripts/run_explorer.sh
Comment (RobbeSneyders) [inline scripts/run_explorer.sh:1]: Is this a duplicate of the `cli.py` script or do we need this for something else?

File: scripts/build_explorer.sh
Comment (RobbeSneyders) [inline scripts/build_explorer.sh:1]: Can you make this file executable?

File: scripts/tag_explorer.sh
Comment (RobbeSneyders) [inline scripts/tag_explorer.sh:1]: Can you make this file executable?

**Review by RobbeSneyders — COMMENTED** [review]
Thanks @ChristiaensBert! Great to have this.

I haven't reviewed the app itself yet, but already left some comments on everything around it.

Some higher level comments:
- This doesn't seem to work with a remote source. Is this something you're still planning to include?
- Can you include documentation on the explorer?

File: fondant/cli.py
Comment (ChristiaensBert) [reply fondant/cli.py:71]: @RobbeSneyders 
Yes, the reason I did this is because `streamlit` always prints a message that says `.... your application can be found on localhost:8501`. But when running the docker container, you can choose another port, e.g. 5000. I thought it might be confusing for the users to see the wrong port, leading to an incorrect link.

But it's indeed not great for debugging. What do you propose?

File: scripts/run_explorer.sh
Comment (ChristiaensBert) [reply scripts/run_explorer.sh:1]: It has the same functionality, I'll delete it

File: fondant/cli.py
Comment (ChristiaensBert) [reply fondant/cli.py:34]: It's a local folder that contains the data (manifest etc), which you mount into the Docker container. I'll think of a better name

File: scripts/tag_explorer.sh
Comment (ChristiaensBert) [reply scripts/tag_explorer.sh:1]: How do you do this? Is it the `set -e`?

File: examples/pipelines/starcoder/pipeline.py
Comment (ChristiaensBert) [reply examples/pipelines/starcoder/pipeline.py:1]: It was autoformatting from black, fixed it

File: scripts/tag_explorer.sh
Comment (ChristiaensBert) [reply scripts/tag_explorer.sh:1]: nvm, it was `chmod u+x FILE`

File: fondant/cli.py
Comment (RobbeSneyders) [reply fondant/cli.py:34]: `--data-directory` is a lot more clear to me :) 

Would be ideal if we can develop a common language instead of referring to the same thing by different names. So would be great if we can align this with fondant. Maybe we should rename `base_path` do `data_directory` (or something similar) there as well @GeorgesLorre @PhilippeMoussalli?

**Comment by ChristiaensBert** [comment]
> Thanks @ChristiaensBert! Great to have this.
> 
> I haven't reviewed the app itself yet, but already left some comments on everything around it.
> 
> Some higher level comments:
> 
> * This doesn't seem to work with a remote source. Is this something you're still planning to include?
> * Can you include documentation on the explorer?

I'll look into the remote source in a next PR! And I added documentation in the README.md of the explorer

**Comment by RobbeSneyders** [comment]
Would be great if we can add it to the docs site as well

**Comment by ChristiaensBert** [comment]
> Would be great if we can add it to the docs site as well

done!

File: fondant/cli.py
Comment (ChristiaensBert) [reply fondant/cli.py:61]: Your call! It would indeed install the Docker library within Fondant, which not everyone might want (if they don't have Docker installed etc). I think this might be fine

**Comment by ChristiaensBert** [comment]
Apparently, the new tests also got merged into this one (after being approved). Sorry for the big PR

File: fondant/cli.py
Comment (ChristiaensBert) [reply fondant/cli.py]: I made both the --credentials and --data-directory optional but with a warning if you don't pass it
```
WARNING:root:You have not provided a data directory.To access local files, provide a local data directory with the --data-directory flag.
WARNING:root:You have not provided a credentials file. If you wish to access data from a cloud provider, mount the credentials file with the --credentials flag.
```

**Comment by RobbeSneyders** [comment]
When using the manifest you mention above, I get
> This file path does not exist

**Comment by ChristiaensBert** [comment]
> When using the manifest you mention above, I get
> 
> > This file path does not exist

@RobbeSneyders 
Did you mount your GCP credentials like this? 

$HOME/.config/gcloud/application_default_credentials.json:/root/.config/gcloud/application_default_credentials.json

See: https://cloud.google.com/docs/authentication/application-default-credentials

**Comment by RobbeSneyders** [comment]
I did not mount my credentials :slightly_smiling_face: Can we get a more relevant error message for this?

**Comment by ChristiaensBert** [comment]
> 
Good point, I added these warnings in the UI

Schermafbeelding 2023-06-16 om 11 37 58

Schermafbeelding 2023-06-16 om 11 38 12

File: docs/data_explorer.md
Comment (GeorgesLorre) [inline docs/data_explorer.md:1]: Is this included in mkdocs ? Maybe a link on the homepage/index

**Review by GeorgesLorre — APPROVED** [review]
There are some changes in this PR that should have been in separate PR's (like the tests and changes to the examples) but overall nice work!

**Pull request merged**
